sdk/core/core-http/lib/proxyAgent.ts

```diff
@@ -15,9 +15,15 @@ export function createProxyAgent(
   proxySettings: ProxySettings,
   headers?: HttpHeaders
 ): ProxyAgent {
+  const host = URLBuilder.parse(proxySettings.host).getHost() as string;
+  if (!host || !isValidPort(proxySettings.port)) {
+    throw new Error(
+      "Expecting a non-empty host and a valid port number in the range of [0, 65535] in proxy settings."
```

Comment:
should we throw two separate errors for each of these validations? It's usually better if errors can be as specific as possible without lumping together two possible failure states

Reply:
I thought about that, but realized that I would have to pick one error to throw, when both host and port number are invalid so maybe one error for two cases is still fine.

Reply:
It's fine to throw one, they fix it, and then the other is thrown. I'd prefer that as a user experience to an ambiguous error message, but maybe I'm in the minority here? @bterlson @ramya-rao-a   thoughts?

Reply:
I agree separate errors is probably better than one umbrella error. No need to unify, throw the first one you notice 😀

Reply:
Updated to throw two separate errors.